stubs/docutils/docutils/writers/latex2e.pyi

```diff
@@ -1,3 +1,21 @@
 from _typeshed import Incomplete
+from collections.abc import Mapping
+from typing import ClassVar
+
+from docutils.utils import Reporter
+
+class Babel:
+    language_codes: ClassVar[Mapping[str, str]]
```

Comment:
Wouldn't it make more sense to use a concrete `dict` here as well as for `active_chars` and `otherlanguages`?

Reply:
i actually had this as a `dict` originally. It's only ever used as a mapping so that seemed the more flexible approach. I'm struggling to think what concrete types subclasses might use for which a concrete type would be too restrictive.

What's your reasoning for using the concrete type?

Reply:
There's no strong reasoning, but just a few general thoughts: A concrete type can be used in some situations where a `Mapping` can't (for example if a function does an isinstance check). That's why concrete types are better suited for output types, while protocols and pseudo-protocols like `Mapping` are better suited for input types. Of course, attributes are both output and input types, so it's a wash. In this case, the attributes look a bit more "outputy" to me, i.e. I wouldn't expect them to be changed normally. But that's no strong reasoning, so I go with what you think is best.

Reply:
Thanks for walking me through your reasoning. All makes sense.

On the basis of that I'm inclined to use a concrete class and see who complains!

Will update

Reply:
done